tests/NumericalUnit/Shape.hs

```diff
@@ -17,10 +17,56 @@ import Test.Hspec
 unitTestShape :: Spec
 unitTestShape = describe "unit tests for Shape" $ do
         specify "foldl on shape" $  S.foldl (+) 0 (1:* 2:* 3 :* Nil )  `shouldBe`   P.foldl   (+) 0  [1,2,3]
+        specify "foldl' on shape" $ S.foldl' (+) 0 (1:* 2:* 3 :* Nil )  `shouldBe`   P.foldl (+) 0  [1,2,3]
         specify "foldr on shape" $  S.foldr (+) 0 (1:* 2:* 3 :* Nil )  `shouldBe`   P.foldr  (+) 0  [1,2,3]
         specify "foldl1 on shape" $  S.foldl1 (+) (1:* 2:* 3 :* Nil )  `shouldBe`   P.foldl1   (+)  [1,2,3]
         specify "foldr1 on shape" $  S.foldr1 (+) (1:* 2:* 3 :* Nil )  `shouldBe`   P.foldr1  (+)  [1,2,3]
 
+        specify "== on Nil" $ Nil == Nil `shouldBe` True
+        specify "== on identical shapes" $ (1:* 2:* 3 :*  Nil) == (1:* 2:* 3 :*  Nil) `shouldBe` True
+        specify "== on different shapes" $ (1:* 2:* 3 :*  Nil) == (1:* 2:* 2 :*  Nil) `shouldBe` False
```

Comment:
i nearly didn't see the 1 2 3 vs the 1 2 2

from a code coverage perspective, even a singleton list / shape should suffice i think

Reply:
also i'd say "shapes with different element", :)